backend/src/test/java/io/swagger/service/TestFilmService.java

```diff
@@ -0,0 +1,191 @@
+package io.swagger.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Random;
+import java.util.UUID;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import io.swagger.entity.Film;
+import io.swagger.repository.FilmDTO;
+import io.swagger.repository.FilmRepository;
+import jakarta.persistence.EntityExistsException;
+
+@ExtendWith(MockitoExtension.class)
+public class TestFilmService {
+
+    private List<Film> mockFilmDB;
+    private int maxFilmLength = 200;
+    private int minFilmLength = 50;
+    private int minYear = 1990;
+    private int maxYear = 2024;
+    private int movieCount = 15;
+
+    @InjectMocks
+    FilmService filmService;
+
+    @Mock
+    FilmRepository filmRepository;
+
+    @BeforeEach
+    public void initData() {
+        mockFilmDB = new ArrayList<Film>();
```

Comment:
Hier vielleicht lieber mit when(...)thenReturn(...)-Syntax arbeiten.

Reply:
Dann macht es erst recht Sinn when...thenReturn... zu verwenden und das dann ggf. auch in den beforeEach-Teil einzubinden.

Reply:
das habe ich in mein individuell Test geschrieben und initData ist nur für mögliche Mockdaten gedacht nd damit auch randomiziert. D.h. randomiziert Daten bei Testing und es muss immer funktioniert.